### PR TITLE
style: Opa AJ style 🕺

### PR DESCRIPTION
**Pull request by coolaj86**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,55 +1,102 @@
...
+
+var BodyParser = require('body-parser');
+var Eta = require('eta');
 var express = require('express');
-var eta = require('eta');
-var bodyParser = require('body-parser');
 var morgan = require('morgan');
 
-var app = express();
+var app = require('@root/async-router').Router();
```
Comment (kaihendry) [inline index.js:14]: app.engine doesn't work on this. It does of course on `var app = express();`

Comment (coolaj86) [reply index.js:14]: You need to add it to your `package.json`:

```js
npm install --save @root/async-router
```

File: index.js [hunk 2]
```diff
@@ -1,55 +1,102 @@
...
+var server = express().use('/', app);
+var urlencodedParser = BodyParser.urlencoded({ extended: false });
 
 app.use(morgan('dev'));
 
-app.engine('eta', eta.renderFile);
+app.engine('eta', Eta.renderFile);
 app.set('view engine', 'eta');
-app.set('json spaces', 2);
```
Comment (coolaj86) [inline index.js:22]: BUG!!! should be `server.engine` and `server.set`.

**Comment by kaihendry** [comment]
I merged it but no _root/async-router_ as yet as I think I'll only bring it when I feel the need.

Thank you ever so much AJ!

**Comment by coolaj86** [comment]
If you follow this pattern in _EVERY_ route, then you'll never need `@root/async-router`:

```js
app.get('/', function (req, res, next) {
    Promise.resolve(async function () {
        // do stuff
    }).catch(next);
});
```

If you don't use that pattern (or @root/async-router), then you can't correctly catch errors because express 5.x was never released, never will be released, and express 4.x has no reasonable way to handle errors.

**Comment by kaihendry** [comment]
I'm still a bit confused about error handling: https://github.com/kaihendry/node-express-starter/commit/2932816f3c1801060a260ce04676ebd663e3825d

**Pull request merged**